cmscontrib/loaders/lio.py

```diff
@@ -0,0 +1,322 @@
+from cmscommon.constants import \
+    SCORE_MODE_MAX, SCORE_MODE_MAX_SUBTASK, SCORE_MODE_MAX_TOKENED_LAST
+from .italy_yaml import load_yaml_from_path, make_timedelta
+from datetime import timedelta
+
+logger = logging.getLogger(__name__)
+
+
+def set_if_present(src_dict, trg_dict, key, conv=None, default=None):
+    if key in src_dict:
+        if conv:
+            trg_dict[key] = conv(src_dict[key])
+        else:
+            trg_dict[key] = src_dict[key]
+    elif default is not None:
+        trg_dict[key] = default
+
+
+class LioTaskLoader(TaskLoader):
+
+    short_name = "Lio-task"
+    description = "Latvian Informatics Olympiad task loader"
+
+    def __init__(self, path, file_cacher):
+        super().__init__(path, file_cacher)
+        self.task_dir = os.path.dirname(self.path)
+        self.conf = load_yaml_from_path(self.path)
+
+
+    @staticmethod
+    def detect(path):
+        # TODO: Support auto detection
+        return False
+
+
+    def parse_point_file(self, point_path):
+        """
+        Parse point file with the following format for each line
+            {from group}-{till group} {points for each group} {comment}
+
+        return ({group: points}): Dictionary of points per group
+        """
+        with open(point_path, "rt", encoding="utf-8") as f:
+            content = [line.strip() for line in f.readlines()]
+        points_per_group = dict()
+        for line in content:
+            vars = line.replace("-", " ").split()
+            a = int(vars[0])
+            b = int(vars[1])
+            points = int(vars[2])
+            for group in range(a, b+1):
+                if group in points_per_group:
+                    logger.critical("Duplicated groups in point file")
+                    raise ValueError
+                points_per_group[group] = points
+        for group in range(len(points_per_group)):
+            if group not in points_per_group:
+                logger.critical("Missing group from point file")
+                raise ValueError
+        return points_per_group
+
+
+    def get_task(self, get_statement):
+        args = {
+            'name': self.conf['name'],
+            'title': self.conf['title'],
+        }
+        name = args['name']
+
+        logger.info(f"Loading parameters for task {name}")
+
+        if get_statement:
+            args['statements'] = {}
+            for statement in self.conf.get('statements', []):
+                path, lang = statement
+                logger.info(f"Loading statement: {statement}")
+                digest = self.file_cacher.put_file_from_path(
+                    os.path.join(self.task_dir, path),
+                    f"Statement for task {name} (lang: {lang})",
+                )
+                args['statements'][lang] = Statement(lang, digest)
+            if args['statements']:
+                args['primary_statements'] = self.conf.get('primary_statements', ["lv"])
+
+        args['submission_format'] = [f"{name}.%l"]
```

Comment:
Note that this doesn't necessarily match the task name. Previously we had, for example, tasks named `KastesJ` and `KastesV`, but the expected submission format and Java class name was `Kastes`. I suppose this necessitates one more field in the YAML file.

Reply:
Added field.